example/particles/particles.cpp

```diff
@@ -169,14 +176,15 @@ TaskStatus DepositParticles(MeshBlock *pmb) {
   return TaskStatus::complete;
 }
 
-TaskStatus CreateSomeParticles(MeshBlock *pmb) {
+TaskStatus CreateSomeParticles(MeshBlock *pmb, double t0) {
   auto pkg = pmb->packages.Get("particles_package");
   auto swarm = pmb->swarm_data.Get()->Get("my particles");
   auto rng_pool = pkg->Param<RNGPool>("rng_pool");
   auto num_particles = pkg->Param<int>("num_particles");
   auto v = pkg->Param<Real>("particle_speed");
 
-  const auto new_particles_mask = swarm->AddEmptyParticles(num_particles);
+  ParArrayND<int> new_indices;
```

Comment:
Yes these arrays are always 1D. But `ParArrayND` comes with some additional nice features like `GetHostMirrorAndCopy()` These are mostly just wrapping Kokkos methods I think, but on some level it's nice to have Kokkos and the execution space stuff abstracted away. That being said I wasn't aware that ParArray1D could be more performant... what kind of performance hit are we talking about?

Reply:
The performant hit shouldn't be substantial. `ParArrayND` and `ParArray1D` are the same performance cost in general. The issue is that `ParArrayND` carries around extra metadata. And if enough metadata is copied during a kernel launch, the launch latency increases in a discontinuous way as `Kokkos` switches from a lightweight to a heavy kernel... I don't fully understand the details. 

But basically this comes up if you copy like of order 10 `ParArrayND`s into a kernel

Reply:
Got it thanks, that's good to know. This may have big implications for the `SwarmDeviceContext` model, since a bunch of data (coordinates and some ParArrays and stuff) is bundled up and copied. Maybe I need to switch to just allocating that on device and providing a pointer from the `Swarm` class... anyway that's for the future

Reply:
How much refactoring do you expect it to be to make it ParArray1D?
I'm asking as we didn't see a significant performance difference for 3D or 4D arrays versus ND arrays, but for a large number of particles I expect a measurable difference between always calculating indices for 6 dimensions versus 1.
Also the `Kokkos::create_mirror_view_and_copy()` works directly for `ParArray1D`s

Reply:
I *think* it shouldn't be too much work to do this, the biggest issue would probably be if variable packs don't like moving away from ParArray1Ds. Since this could end up being a performance issue I'll take a look now.